sdk/iot/hub/src/az_iot_hub_client.c

```diff
@@ -101,6 +104,77 @@ AZ_NODISCARD az_result az_iot_hub_client_get_user_name(
   return AZ_OK;
 }
 
+AZ_NODISCARD az_result az_iot_hub_client_get_user_name_with_model_id(
```

Comment:
Please open a GitHub issue (and put // TODO linking back to GitHub # in the code) calling out need to long-term reconcile this once PnP goes GA.  

The options would be (A) leaving this API as is, but at minimum removing the preview API version stuff or (B) removing this completely and using a PnP SDK instead.

Reply:
Opened #812